runtime/bcutil/cfreader.c

```diff
@@ -2195,18 +2195,35 @@ checkClass(J9PortLibrary *portLib, J9CfrClassFile* classfile, U_8* segment, U_32
 		goto _errorFound;
 	}
 
+	/* Java 9 spec: If the ACC_INTERFACE flag is not set, ACC_MODULE may not be set. */
+	if ((flags & BCT_MajorClassFileVersionMask) >= BCT_Java9MajorVersionShifted) {
+		if ((value & CFR_ACC_MODULE) && (! (value & CFR_ACC_INTERFACE))) {
```

Comment:
Since this is new code, please use the J9 BITS macros.

Reply:
Doesn't this conflict with the below check that if CFR_ACC_INTERFACE is set then CFR_ACC_MODULE cannot be set?

Reply:
Yes - I'll these checks are correct. I've now simplified them into one check.